lsst-epo/investigations-client
# [F] Add number question + validation

## What this change does ##

Adds a new question type for numbers, and an answer validation setup for number questions that can be extended to others.

Resolves #159 

## Notes for reviewers ##

Client + API branch

Include an indication of how detailed a review you want on a 1-10 scale.
- 5 = "Please make sure there are no obvious errors and that you believe it does what it says it does"

## Testing ##

- Create a new number type question in Craft, the first validation fields (minimum, maximum, precision) are attached to the `input` element
- Add validator fields, an example is shown below. For very large numbers abbreviations can be used, the comparison value will be parsed as LaTeX
![image](https://github.com/lsst-epo/investigations-client/assets/11721838/c135e1a5-fa29-4ae6-bb1c-6ee6b00c731f)
- Add question to a page
- View the question in the client, begin input. Validations will be run on blur event so as not to barrage users with notifications on every change.


## Checklist ##

If any of the following are true, please check them off

## Screenshots (optional) ##
### Before:
### After:
